From 4bc79153290b7a58af9876d7ad87eaf8219ac304 Mon Sep 17 00:00:00 2001
From: Wendy Liang <wendy.liang@xilinx.com>
Date: Mon, 17 Aug 2020 14:31:59 -0700
Subject: [PATCH 1522/1852] misc: xilinx-ai-engine: gate columns first in
 partition reset

commit 3beab0ddef7f3a6f602023306d33244356af6915 from
https://github.com/Xilinx/linux-xlnx.git

It can cause ECC or parity errors if reset the partition while there is
any write transaction in progress. To avoid this issue, will need to gate
the columns before reset the columns.

Signed-off-by: Wendy Liang <wendy.liang@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/misc/xilinx-ai-engine/ai-engine-reset.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/misc/xilinx-ai-engine/ai-engine-reset.c b/drivers/misc/xilinx-ai-engine/ai-engine-reset.c
index b67ef5e60e50..d35cd8d17d95 100644
--- a/drivers/misc/xilinx-ai-engine/ai-engine-reset.c
+++ b/drivers/misc/xilinx-ai-engine/ai-engine-reset.c
@@ -128,6 +128,7 @@ static void aie_part_clear_mems(struct aie_partition *apart)
  * @return: 0 for success and negative value for failure
  *
  * This function will:
+ *  * gate all the columns
  *  * reset AI engine partition columns
  *  * reset AI engine shims
  *  * clear the memories
@@ -144,6 +145,7 @@ int aie_part_clean(struct aie_partition *apart)
 	if (apart->cntrflag & XAIE_PART_NOT_RST_ON_RELEASE)
 		return 0;
 
+	aie_part_set_cols_clkbuf(apart, false);
 	aie_part_set_cols_reset(apart, true);
 
 	ret = apart->adev->ops->reset_shim(adev, &apart->range);
-- 
2.31.1

